From 9b13db401a2c1a6764736ea8d11b009a9f43464a Mon Sep 17 00:00:00 2001
From: rmunjewar <rmunjewar@maxlinear.com>
Date: Thu, 4 Jul 2024 15:14:48 +0530
Subject: [PATCH] WLANRTSYS-82700 : iwlwav-hostap-ng : porting hostap to
 hostap-ng, optimise acs scan

Porting changes for commit
eb087691cbb	WLANRTSYS-80165	Optimized scan for AFC support in ACS WAV700 in 6GHz

FIXUP for 	WLANRTSYS-50170	Smart ACS implementation

Signed-Off-By : Rushabh Munjewar <rmunjewar@maxlinear.com>
---
 src/common/mxl_ieee802_11_common.c | 14 ++++++++++++++
 src/common/mxl_ieee802_11_common.h |  1 +
 2 files changed, 15 insertions(+)

diff --git a/src/common/mxl_ieee802_11_common.c b/src/common/mxl_ieee802_11_common.c
index cdf624579..a33f03eeb 100644
--- a/src/common/mxl_ieee802_11_common.c
+++ b/src/common/mxl_ieee802_11_common.c
@@ -14,6 +14,7 @@
 #include "utils/includes.h"
 #include "utils/list.h"
 #include "utils/common.h"
+#include "ieee802_11_common.h"
 #include "mxl_ieee802_11_common.h"
 
 #define MLO_TYPE_BASIC 0
@@ -140,3 +141,16 @@ int mxl_parse_optional_neighbor_fields(char **tmp_o, const char **end_o, u8 **ne
 
 	return 0;
 }
+
+/**
+ * mxl_is_6ghz_unii5_unii7 - Check if the frequency belongs to UNII5 or UNII7 6GHz Band
+ * @freq: frequency to check
+ */
+bool mxl_is_6ghz_unii5_unii7(int freq)
+{
+	if (is_6ghz_freq(freq)) {
+		if ((5925 <= freq && freq <= 6425) || (6525 <= freq && freq <= 6875))
+			return true;
+	}
+	return false;
+}
diff --git a/src/common/mxl_ieee802_11_common.h b/src/common/mxl_ieee802_11_common.h
index 44ee4077d..4c526d836 100644
--- a/src/common/mxl_ieee802_11_common.h
+++ b/src/common/mxl_ieee802_11_common.h
@@ -19,6 +19,7 @@
 void mxl_scan_res_set_rssi_threshold(struct wpa_scan_results *scan_res, int rssi_threshold);
 bool mxl_scan_res_bss_below_threshold(struct wpa_scan_results *scan_res, int bss_rssi);
 int mxl_parse_optional_neighbor_fields(char **tmp_o, const char **end_o, u8 **nei_pos_o, const char **pos_o, u8 **nei_rep_o, size_t nei_rep_len);
+bool mxl_is_6ghz_unii5_unii7(int freq);
 
 inline int action_payload_min_len_expected(u8 *action)
 {
-- 
2.43.0

